### PR TITLE
Feature/redbox 146 make file subapp

### PR DESCRIPTION
**Pull request by rachaelcodes**
## Context
`app.py` in `core_api` was getting quite large, and would only extend by building the chat app. I wrote an earlier PR to address this, but this is a simpler approach.

## Changes proposed in this pull request
The `/file` endpoints are moved to a sub-application (in its own subapp), and then mounted on the root app.

Annoyingly, the documentation for these endpoints is now moved to `/file/docs`. We could have a future ticket to create an endpoint documenting all possible endpoints for the core_api, e.g. https://github.com/tiangolo/fastapi/issues/3767#issuecomment-905633830 

## Guidance to review

Check the endpoints are split out as expected. 

## Relevant links
https://technologyprogramme.atlassian.net/jira/software/projects/REDBOX/boards/409?selectedIssue=REDBOX-146

## Things to check

- [ ] I have added any new ENV vars in all deployed environments
- [x] I have tested any code added or changed
- [ ] I have run integration tests
- [ ] I have manually tested the streamlit app

### CODE REVIEW

File: core_api/Dockerfile [hunk 1]
```diff
@@ -25,7 +25,7 @@ ENV VIRTUAL_ENV=/app/.venv \
     PATH="/app/.venv/bin:$PATH"
 COPY --from=builder ${VIRTUAL_ENV} ${VIRTUAL_ENV}
 
-ADD core_api/src/app.py /app/app.py
+COPY core_api/ /app/core_api/
```
Comment (gecBurton) [inline core_api/Dockerfile:28]: :+1:

File: core_api/src/routes/file.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    file = File(
+        name=name,
+        url=str(location),  # avoids JSON serialisation error
+        content_type=type,
+    )
+
+    storage_handler.write_item(file)
+
+    log.info(f"publishing {file.uuid}")
+    await router.broker.connect()
```
Comment (gecBurton) [inline core_api/src/routes/file.py:72]: does this need to happen once as part of setup, or even within https://fastapi.tiangolo.com/advanced/events/?

Comment (lmwilkigov) [reply core_api/src/routes/file.py:72]: I think moving it above would be a lot cleaner network IO wise

**Review by gecBurton — CHANGES_REQUESTED** [review]
one RFC, otherwise LGTM!

File: core_api/src/routes/file.py (hunk 2 above)
Comment (rachaelcodes) [reply core_api/src/routes/file.py:72]: Unfortunately, the setup only happens on the root app, not any sub ones (https://github.com/tiangolo/fastapi/issues/1480#issuecomment-636317883 )

Comment (gecBurton) [reply core_api/src/routes/file.py:72]: @lmwilkigov after speaking to @rachaelcodes I see that this is not straightforward. I am going to approve this and raise a tech debt ticket to come back to this.

**Review by gecBurton — APPROVED** [review]
lets address the connection of the broker in another ticket

File: core_api/src/routes/file.py (hunk 2 above)
Comment (gecBurton) [reply core_api/src/routes/file.py:72]: https://technologyprogramme.atlassian.net/browse/REDBOX-154

**Pull request merged**